include/simdjson/numberparsing.h

```diff
@@ -572,17 +615,23 @@ static really_inline bool parse_number(const uint8_t *const buf, ParsedJson &pj,
         // Ok, chances are good that we had an overflow!
         // this is almost never going to get called!!!
         // we start anew, going slowly!!!
-        return parse_float(buf, pj, offset, found_minus);
+        return parse_long_float(buf, pj, offset, found_minus, first_after_period ?: digit_end, exp_number, digit_end, p);
       }
     }
     if (unlikely((power_index > 2 * 308))) { // this is uncommon!!!
       // this is almost never going to get called!!!
-      // we start anew, going slowly!!!
-      return parse_float(buf, pj, offset, found_minus);
+      return parse_float_strtod(buf, pj, offset, p);
+    }
+    double d = compute_float_double(i, exponent, power_index, negative);
```

Comment:
so this would now do a div, even  if the power is negative. the speed difference didn't seem very big so i figured i'd show this and i can revert if we want

Reply:
s/even if/if/